insights/terraform/projects/bfd/api-requests/lambda_src/bfd-cw-to-flattened-json.py

```diff
@@ -0,0 +1,292 @@
+the 6000000 limit meets your requirements. If your data contains a sizable number of records that are classified as
+Dropped/ProcessingFailed, then it is suggested to lower the 6000000 limit within the function to a smaller value
+(eg: 5000000) in order to confine to the 6MB (6291456 bytes) payload limit imposed by lambda. You can find Lambda
+quotas at https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-limits.html
+
+The code below will:
+
+1) Gunzip the data
+2) Parse the json
+3) Set the result to ProcessingFailed for any record whose messageType is not DATA_MESSAGE, thus redirecting them to the
+   processing error output. Such records do not contain any log events. You can modify the code to set the result to
+   Dropped instead to get rid of these records completely.
+4) For records whose messageType is DATA_MESSAGE, extract the individual log events from the logEvents field, and pass
+   each one to the transformLogEvent method. You can modify the transformLogEvent method to perform custom
+   transformations on the log events.
+5) Concatenate the result from (4) together and set the result as the data of the record returned to Firehose. Note that
+   this step will not add any delimiters. Delimiters should be appended by the logic within the transformLogEvent
+   method.
+6) Any individual record exceeding 6,000,000 bytes in size after decompression and encoding is marked as
+   ProcessingFailed within the function. The original compressed record will be backed up to the S3 bucket
+   configured on the Firehose.
+7) Any additional records which exceed 6MB will be re-ingested back into Firehose.
+8) The retry count for intermittent failures during re-ingestion is set 20 attempts. If you wish to retry fewer number
+   of times for intermittent failures you can lower this value.
+"""
+
+"""
+Except where noted as a 'BFD modification' all code here is original to the AWS
+Blueprint and should not be changed without accepting the maintenance burden.
+"""
+
+import base64
+import json
+import gzip
+from io import BytesIO
+import boto3
+
+
+"""
+BFD modification to add a function to flatten a JSON object (but not pivot out arrays).
+Code credits: https://towardsdatascience.com/flattening-json-objects-in-python-f5343c794b10
+"""
+def flatten_json(y):
+    out = {}
+
+    def flatten(x, name=''):
+        if type(x) is dict:
+            for a in x:
+                flatten(x[a], name + a + '.')
+        else:
+            out[name[:-1]] = x
+
+    flatten(y)
+    return out
+
+
+
+def transformLogEvent(log_event):
+    """Transform each log event.
+
+    The default implementation below just extracts the message and appends a newline to it.
+
+    Args:
+    log_event (dict): The original log event. Structure is {"id": str, "timestamp": long, "message": str}
+
+    Returns:
+    str: The transformed log event.
+    """
+    
+    """
+    BFD modification to the blueprint to flatten the message json.
+    V1
```

Comment:
this is silly but I put in this 'V1' line while trying to test that file differences were detected. would you mind removing it?